src/main/webapp/app/lecture/lecture.route.ts

```diff
@@ -12,14 +12,14 @@ import { Lecture } from 'app/entities/lecture.model';
 import { LectureAttachmentsComponent } from 'app/lecture/lecture-attachments.component';
 import { Authority } from 'app/shared/constants/authority.constants';
 import { CourseResolve } from 'app/course/manage/course-management.route';
+import { lectureUnitRoute } from 'app/lecture/lecture-unit/lecture-unit-management/lecture-unit-management.route';
 
 @Injectable({ providedIn: 'root' })
 export class LectureResolve implements Resolve<Lecture> {
     constructor(private service: LectureService) {}
 
     resolve(route: ActivatedRouteSnapshot): Observable<Lecture> {
-        // TODO: This should always use lectureId and never just 'id'
-        const id = route.params['lectureId'] ? route.params['lectureId'] : route.params['id'] ? route.params['id'] : undefined;
+        const id = route.params['lectureId'] ? route.params['lectureId'] : undefined;
```

Comment:
I think this can be simplified to
```suggestion
        const id = route.params['lectureId'];
```
because `route.params['lectureId']` will be `undefined` anyway if the `lectureId` property doesn't exist.